src/wayland_frontend/engine.rs

```diff
@@ -52,8 +53,26 @@ impl Engine {
     pub fn new(coordinator: Coordinator,
                settings: Settings,
                keyboard_config: KeyboardConfig) -> Self {
+        let mut partial_socket_path = PathBuf::from(std::env::var("XDG_RUNTIME_DIR").unwrap_or("/run/user/1000".to_owned()));
```

Comment:
I forgot to write it in issue, but the order to check should be:
 - default socket
 - wayland-{} sockets
I know you try wayland-0 anyway but what wl::DisplaySocket::new_default does is it's private matter and its behavior may change (theoretically).

I don't like hardcoding the path. I will most probably fail anyway. I would exit if the variable is not set. Specification [1] says "it MUST be created when the user first logs in" so it's bug in desktop environment if it is not set.

[1] https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html